src/PhpSpec/Matcher/FloatValueMatcher.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+/*
+ * This file is part of PhpSpec, A php toolset to drive emergent
+ * design by specification.
+ *
+ * (c) Marcello Duarte <marcello.duarte@gmail.com>
+ * (c) Konstantin Kudryashov <ever.zet@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace PhpSpec\Matcher;
+
+use PhpSpec\Exception\Example\FailureException;
+use PhpSpec\Exception\Example\NotEqualException;
+use PhpSpec\Formatter\Presenter\Presenter;
+
+final class FloatValueMatcher extends BasicMatcher
+{
+    /**
+     * @var array
+     */
+    private static $keywords = [
+        'return',
+        'be',
+        'equal',
+        'beEqualTo'
+    ];
+
+    /**
+     * @var Presenter
+     */
+    private $presenter;
+
+    /**
+     * @param Presenter $presenter
+     */
+    public function __construct(Presenter $presenter)
+    {
+        $this->presenter = $presenter;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function supports(string $name, $subject, array $arguments): bool
+    {
+        return \is_float($subject)
+            && \in_array($name, self::$keywords)
+            && 1 === \count($arguments)
+            && \is_float($arguments[0]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getPriority(): int
+    {
+        return 150;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function matches($subject, array $arguments): bool
+    {
+        if (extension_loaded('bcmath')) {
+            return \bccomp($subject, $arguments[0], 20) === 0;
```

Comment:
I wasn't sure what value to use for the `scale` parameter. Whatever we use it will have to be documented.